chore: remove more unused code

incendial opened the pull request:
After the mention [here](https://github.com/flutter/devtools/pull/5918#discussion_r1232529612) I ran the unused check again with a different flag and got a bunch of new highlights. Some of them look illegal 😅.  

Note: not all of the removed code might be unused intentionally, but I have no context of that. Also, there are might be not obvious to me false-positives.

RELEASE_NOTE_EXCEPTION=[Internal refactoring]

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [-] I listed at least one issue that this PR fixes in the description above.
- [-] I updated/added relevant documentation (doc comments with `///`).
- [-] I added new tests to check the change I am making, or there is a reason for not adding tests.

incendial commented:
@jacob314 @polina-c fyi

incendial commented:
Hm, I didn't notice that `devtools_shared` is a package you publish to pub, so I expect some of them to be reverted, but waiting for the review.

polina-c commented:
Looks good. Some questions:

1. Can you please test that Memory > Chart is still functioning?
2. What is our process around removing unused code? Is there any continuous automation that kicks it off?

polina-c commented:
@elliette @kenzieschmoll @CoderDake 

Can you also review?

incendial commented:
> What is our process around removing unused code?

You mean you want to run `dcm unused code` checks on bots?

incendial commented:
Here is the video of working app:

https://github.com/flutter/devtools/assets/21116742/5c61f472-b8f4-402c-a5fb-ec583763843e

Looks like everything works except for one overflow error (but I'm not sure whether it existed before):

https://github.com/flutter/devtools/assets/21116742/1a36695e-8e18-4345-93ed-aa0543d804b6

polina-c commented:
LGTM assuming others LGTM their parts

CoderDake commented on packages/devtools_app/lib/src/shared/eval_on_dart_library.dart:
```diff
@@ -657,13 +641,11 @@ class CancelledException implements Exception {}
-  final Map<String, String?>? scope;
 
   @override
   String toString() {
```
probably better to add scope to toString

CoderDake commented on packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart:
```diff
@@ -69,7 +52,6 @@ class SourcePosition {
     );
   }
 
-  final String? file;
```
it might be better to add this to toString

incendial replied:
Sure, do you have any preference regarding the format (ex. '$file:$line:$column' or '$file\n$line:$column')?

kenzieschmoll replied:
'$file:$line:$column'  SGTM

incendial replied:
What bothers me is that file is not passed anywhere and is not used in `bool operator ==(other) {` and `int get hashCode =>`. On top of that any `toString` call will start with `null:`.

Are you sure this field should be restored?

kenzieschmoll replied:
@elliette does this seem like a bug that we are not using the `file` parameter anywhere? I suspect this would cause weird equality issues where two `SourcePosition` objects appear identical because they have the same `line` & `column` & `tokenPos` but they are actually from two different files.

elliette replied:
I don't think this is a bug. It looks like `file` parameter was added in PR https://github.com/flutter/devtools/pull/3448 to combine two definitions of `SourcePosition` in two different files (`debugger_model` and `diagnostics_node`), the second of which had `file` as a parameter but never used it. 

`SourcePosition` is usually used in conjunction with some other class that contains file information (for example `ScriptRef` in [`StackFrameAndSourcePosition`](https://github.com/flutter/devtools/blob/6a2271a81ecea41aeb77ffed617587f1cceccc3d/packages/devtools_app/lib/src/screens/debugger/debugger_model.dart#L189)).

kenzieschmoll replied:
Okay thanks for the input - @incendial why don't we just remove this parameter completely then as you originally did.

CoderDake reviewed (commented):
I have some concerns about devtools_shared and some other small comments

kenzieschmoll reviewed (commented):
please remove all changes to `packages/devtools_shared`. The `devtools_shared` package is published and is used by the devtools server, which is part of the Dart SDK.

kenzieschmoll commented on packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_context_menu.dart:
```diff
@@ -15,14 +14,11 @@ class InstanceViewWithContextMenu extends StatelessWidget {
 
   final int count;
   final MenuBuilder? menuBuilder;
-  final MemoryAreas gaContext;
-
```
@polina-c this seems like a bug that this is unused?

incendial replied:
Should I add a todo or can you suggest what changes are expected to fix it?

kenzieschmoll replied:
@polina-c please advise

incendial replied:
I see that this has already been addressed, I'm not adding it then

kenzieschmoll commented on packages/devtools_app/lib/src/screens/memory/shared/primitives/memory_timeline.dart:
```diff
@@ -117,16 +84,6 @@ class MemoryTimeline {
     _eventSamples.add(event);
   }
 
-  void addSnapshotEvent({bool auto = false}) {
```
@polina-c this also seems like a bug that we have stopped adding an event to the timeline when we take a snapshot. Maybe the action item here is to use this code.

incendial replied:
Same as https://github.com/flutter/devtools/pull/5949#discussion_r1342165085

kenzieschmoll replied:
@polina-c please advise

incendial replied:
This one as well

kenzieschmoll commented on packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart:
```diff
@@ -355,14 +355,6 @@ class CpuProfileData {
-  /// vmTag [tag].
-  ///
-  /// [originalData] does not need to be [processed] to run this operation.
-  factory CpuProfileData.fromVMTag(CpuProfileData originalData, String tag) {
```
@bkonyi is this a bug that this code is unused?

incendial replied:
Same as https://github.com/flutter/devtools/pull/5949#discussion_r1342165085

kenzieschmoll replied:
@bkonyi please advise

incendial replied:
Waiting for this one and then will address all latest comments and bump the version.

bkonyi replied:
Sorry for the delay on this.

I think it can be removed, it was probably added for symmetry and then never used.

kenzieschmoll commented on packages/devtools_app/lib/src/service/service_extensions.dart:
```diff
@@ -57,7 +57,6 @@ class ServiceExtensionDescription<T> {
   ServiceExtensionDescription({
     this.iconAsset,
     this.iconData,
-    List<String>? displayValues,
```
can you revert changes to this file? we have an outstanding issue to add the UI control for the `togglePlatformMode` service extension, at which point this would need to be added back. https://github.com/flutter/devtools/issues/2780

incendial replied:
Done

kenzieschmoll commented on packages/devtools_app/lib/src/shared/diagnostics/diagnostics_node.dart:
```diff
@@ -8,7 +8,6 @@ import 'package:collection/collection.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/rendering.dart';
-import 'package:vm_service/vm_service.dart';
```
I recommend that we remove the changes to this file. This class is extended and used by a customer inside google, so we do not want to risk breaking them. Also, we some of these helpers will be useful as we continue to evolve the Inspector and Layout Explorer and add advanced functionality.

incendial replied:
Sure, done.

kenzieschmoll commented on packages/devtools_app/lib/src/shared/diagnostics/inspector_service.dart:
```diff
@@ -673,32 +625,6 @@ abstract class InspectorObjectGroupBase
           );
   }
 
-  Future<RemoteDiagnosticsNode?> invokeServiceMethodWithArgReturningNode(
-    String methodName,
-    String arg,
-  ) async {
-    if (disposed) return null;
-    return useDaemonApi
-        ? parseDiagnosticsNodeDaemon(
-            invokeServiceMethodDaemonArg(methodName, arg, groupName),
-          )
-        : parseDiagnosticsNodeObservatory(
-            invokeServiceMethodObservatoryWithGroupName1(methodName, arg),
-          );
-  }
```
this method is used by an internal customer. Please revert.

incendial replied:
Done

kenzieschmoll commented on packages/devtools_app/lib/src/shared/memory/adapted_heap_data.dart:
```diff
@@ -62,9 +61,8 @@ class AdaptedHeapData {
+    HeapSnapshotGraph graph,
+  ) async {
     final objects = <AdaptedHeapObject>[];
     for (final i in Iterable.generate(graph.objects.length)) {
```
it looks like the graph parameter is being used here, so is this removal a false positive?

incendial replied:
Only the isolateId was removed here (from 2 parameters `graph and `isolateId` to only `graph` remaining).

Hixie commented:
@incendial Thanks for your contribution! Is this still on your radar?

incendial commented:
@Hixie hi, thanks for the reminder, I hope to get back to this PR soon.

Btw, any chance we can discuss whether the Flutter team could benefit from using Dart Code Metrics?

Hixie commented:
@incendial There's been some discussion of that, I forget the conclusion. I recommend bringing it up on the #hackers-framework channel of our [Discord](https://github.com/flutter/flutter/wiki/Chat).

incendial commented on packages/devtools_app/lib/src/shared/eval_on_dart_library.dart:
```diff
@@ -625,18 +617,10 @@ class EvalOnDartLibrary extends DisposableController
-
-  final String name;
-
-  String get message => 'Library matchining $name not found';
```
Done

incendial commented on packages/devtools_app/lib/src/shared/eval_on_dart_library.dart:
```diff
@@ -701,13 +681,11 @@ class EvalSentinelException extends SentinelException {
-  final Map<String, String?>? scope;
 
   @override
   String toString() {
```
Done for all 3, although not sure whether the scope should go last. Can you please take a look?

CoderDake replied:
LGTM :D

kenzieschmoll reviewed (commented):
please add a changelog entry for package:devtools_app_shared, which is also published on pub. This package is still under early development so these breaking changes are okay. Please put the changes under a new entry for version 0.0.6 and bump the package version in the pubspec.yaml to 0.0.6

kenzieschmoll commented on packages/devtools_app/lib/src/shared/primitives/utils.dart:
```diff
@@ -435,24 +432,6 @@ class JsonUtils {
-  }
-}
-
-/// Add pretty print for a JSON payload.
-extension JsonMap on Map<String, Object?> {
-  String prettyPrint() => const JsonEncoder.withIndent('  ').convert(this);
```
please add this extension method back as this is useful for debugging purposes

incendial commented:
@kenzieschmoll I've updated the package version to 0.0.7 and updated the changelog. Can you please take a look?
The only unresolved conversation is this one https://github.com/flutter/devtools/pull/5949#discussion_r1367683654. Are you sure you want this field back?

And as for failing pipelines, looks like they are failing not because of the changes introduced here. The commit before I upstreamed the main branch got all checks passed.

kenzieschmoll commented:
> And as for failing pipelines, looks like they are failing not because of the changes introduced here. The commit before I upstreamed the main branch got all checks passed.

If you merge master, these issues should be resolved.

incendial commented:
@kenzieschmoll I think all threads are now address + the pipelines are green. Can you please take a final look? Hope we can merge this today 😁
